### PR TITLE
feat: prepare for breaking change in Protobuf C++ API

### PR DESCRIPTION
**Pull request by evalon32**
Protobuf 6.30.0 will change the return types of Descriptor::name() and other methods to absl::string_view. This makes the code work both before and after such a change.

Commit Message: feat: prepare for breaking change in Protobuf C++ API
Additional Description: This makes the implementation compatible with a future Protobuf release where various methods return absl::string_view instead of std::string or const std::string&.
Risk Level: Low
Testing: unit tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Review by yanavlasov — DISMISSED** [review]

**Comment by yanavlasov** [comment]
@evalon32  can you merge main to see if it fixes permafail in the fuzz_coverage infra, please?

**Comment by yanavlasov** [comment]
Ah I see. Please adjust expected coverage value here: https://github.com/envoyproxy/envoy/blob/f918986198223bc707c01a230242ad136012e677/test/per_file_coverage.sh#L35

Lowering expected coverage % in this case is OK, since the change does not actually affect coverage, just reformats lines to cause the ratio to change.

/wait

File: test/per_file_coverage.sh [hunk 1]
```diff
@@ -32,7 +32,7 @@ declare -a KNOWN_LOW_COVERAGE=(
 "source/extensions/filters/common/fault:94.5"
 "source/extensions/filters/common/rbac:90.8"
 "source/extensions/filters/http/cache:95.9"
-"source/extensions/filters/http/grpc_json_transcoder:94.2" # TODO(#28232)
+"source/extensions/filters/http/grpc_json_transcoder:94.0" # TODO(#28232)
```
Comment (alyssawilk) [inline test/per_file_coverage.sh:35]: :-(

**Review by alyssawilk — APPROVED** [review]
Thanks for doing the clean up allowing the bump!

File: test/per_file_coverage.sh (hunk 1 above)
Comment (yanavlasov) [reply test/per_file_coverage.sh:35]: Not ideal. The PR itself does not change coverage at all, just code reformat alters coverage ratio.

**Pull request merged**